src/QueryClause.ts

```diff
@@ -0,0 +1,334 @@
+/**
+ * Base QueryClause implementation.
+ */
+export abstract class QueryClause {
+    abstract toJson(): any;
```

Comment:
@shinsuke-nara At lease as I could find the documents from internet, there seems not these kind of feature in typescript...
Let us keep the current design, we can change it later if there is better solution.